mysql/Dockerfile

```diff
@@ -1,19 +1,47 @@
 FROM fedora
-MAINTAINER http://fedoraproject.org/wiki/Cloud
 
-RUN dnf -y update && dnf clean all
-RUN dnf -y install community-mysql-server community-mysql pwgen supervisor bash-completion psmisc net-tools && dnf clean all
+# MySQL image for OpenShift.
+#
+# Volumes:
+#  * /var/lib/mysql/data - Datastore for MySQL
+# Environment:
+#  * $MYSQL_USER - Database user name
+#  * $MYSQL_PASSWORD - User's password
+#  * $MYSQL_DATABASE - Name of the database to create
+#  * $MYSQL_ROOT_PASSWORD (Optional) - Password for the 'root' MySQL account
 
-ADD ./start.sh /start.sh
-ADD ./config_mysql.sh /config_mysql.sh
-ADD ./supervisord.conf /etc/supervisord.conf
+MAINTAINER http://fedoraproject.org/wiki/Cloud
 
-# RUN echo %sudo	ALL=NOPASSWD: ALL >> /etc/sudoers
+ENV MYSQL_VERSION=10.0 \
+    HOME=/var/lib/mysql
 
-RUN chmod 755 /start.sh
-RUN chmod 755 /config_mysql.sh
-RUN /config_mysql.sh
+LABEL io.k8s.description="MySQL is a multi-user, multi-threaded SQL database server" \
+      io.k8s.display-name="MySQL 5.6" \
+      io.openshift.expose-services="3306:mysql" \
+      io.openshift.tags="database,mysql,mysql56"
 
 EXPOSE 3306
 
-CMD ["/bin/bash", "/start.sh"]
+# This image must forever use UID 27 for mysql user so our volumes are
+# safe in the future. This should *never* change, the last test is there
+# to make sure of that.
+RUN dnf -y --setopt=tsflags=nodocs install gettext hostname bind-utils community-mysql-server community-mysql && \
+    dnf clean all && \
+    mkdir -p /var/lib/mysql/data && chown -R mysql.0 /var/lib/mysql && \
+    test "$(id mysql)" = "uid=27(mysql) gid=27(mysql) groups=27(mysql)"
```

Comment:
Right. But how does the reader know whether 27 was a dynamically allocated id - which would be fragile and ugly - or statically allocated by Fedora? I only know because I looked into the ranges for a different reason (commit message https://github.com/fedora-cloud/Fedora-Dockerfiles/pull/191/commits/e8f59804e316a6fed8b71f5c72d458c55a86b00a).